src/main/java/com/iota/iri/network/TCPNeighbor.java

```diff
@@ -84,11 +84,14 @@ public void setSink(Socket sink) {
      */
     @Override
     public void send(DatagramPacket packet) {
-        if (sendQueue.remainingCapacity() == 0) {
-            sendQueue.poll();
+        synchronized (sendQueue) {
```

Comment:
That is the quickest solution. Kinda patchy but this is the best thing to do right now.
@akashgoswami is currently working on a new implementation of networking which will solve these issues